crates/net/network/src/transactions.rs

```diff
@@ -992,7 +994,18 @@ impl Future for GetPooledTxRequestFut {
         let mut req = self.as_mut().project().inner.take().expect("polled after completion");
         match req.response.poll_unpin(cx) {
             Poll::Ready(result) => {
-                Poll::Ready(GetPooledTxResponse { peer_id: req.peer_id, result })
+                let request_hashes: Vec<TxHash> = match &result {
+                    Ok(Ok(pooled_txs)) => {
+                        pooled_txs.0.iter().map(|tx_elem| *tx_elem.hash()).collect()
+                    }
+                    _ => Vec::new(),
+                };
```

Comment:
this needs to return the stored requested hashes of the `GetPooledTxResponse` type instead

Reply:
so you're saying we should add `requested_hashes: Vec<TxHash>` to `GetPooledTxRequest` in addition to `GetPooledTxResponse`?